src/Scripting/CSharp/CSharpScript.cs

```diff
@@ -232,6 +256,113 @@ protected override string FormatDiagnostic(Diagnostic diagnostic, CultureInfo cu
         {
             return CSharpDiagnosticFormatter.Instance.Format(diagnostic, culture);
         }
+
+        internal delegate SourceText SourceFileReader(CommandLineSourceFile file, IList<DiagnosticInfo> diagnostics, CommonMessageProvider messageProvider, out string normalizedFilePath);
+        internal static SourceText DefaultFileReader(CommandLineSourceFile file, IList<DiagnosticInfo> fileReadDiagnostics, CommonMessageProvider messageProvider, out string normalizedFilePath)
+        {
+            return CommonCompiler.ReadFileContent(
+                file,
+                fileReadDiagnostics,
+                messageProvider,
+                encoding: null,
+                checksumAlgorithm: Text.SourceHashAlgorithm.Sha1, // TODO: Should we be fetching the checksum algorithm from somewhere?
+                normalizedFilePath: out normalizedFilePath);
+        }
+
+        internal static SyntaxTree[] GetSyntaxTrees(
```

Comment:
Does this depend on T? If not, could it moved to a non-generic class helper class?